### PR TITLE
Fixed parsing for WALL action, now all four coordinates and selected units should be returned correctly

### PR DESCRIPTION
**Pull request by qwyt**
Previously only the starting position was parsed.

Tests with all recording files available in test also added a new recording. 

(The new white spaces and empty lines are based on PEP styles guidelines I can revert them).

### CODE REVIEW

File: tests/test_files.py [hunk 1]
```diff
@@ -32,5 +32,6 @@ def test_files_full(self):
         parse_file_full('tests/recs/de-13.07.aoe2record')
 
     def test_files_fast(self):
+        parse_file_fast("tests/recs/small.mgz")
```
Comment (happyleavesaoc) [inline tests/test_files.py:35]: This line should not be required

File: mgz/fast/__init__.py [hunk 2]
```diff
@@ -178,8 +180,21 @@ def parse_action(action_type, data):
         object_id, = struct.unpack_from('<3xI', data)
         return dict(object_ids=[object_id])
     if action_type == Action.WALL:
-        player_id, x, y = struct.unpack_from('<x3b', data)
-        return dict(player_id=player_id, x=x, y=y)
+        selection_count, player_id, x_start, y_start, x_end, y_end, z0, building_id, z1, const = struct.unpack_from(
```
Comment (happyleavesaoc) [inline mgz/fast/__init__.py:183]: Parse like `selection_count, player_id, x_start, y_start, x_end, y_end, building_id = struct.unpack_from('<6bxh')` and then we don't have to name unused bytes

**Comment by happyleavesaoc** [comment]
Couple minor things; will merge when updated!

File: mgz/fast/__init__.py (hunk 2 above)
Comment (qwyt) [reply mgz/fast/__init__.py:183]: Tested this again with .aoe2recording and .mgz tess files and I'm a bit confused now, the data format in  DE recs seems to be different than in mgz files, e.g:

from 'tests/recs/small.mgz'
sel.   p  x1. y1. x2.  y2.        (building_id)
04   02 3b 2d 3b  2d   00       48 00         00 00 ff ff ff ff 0b 08 00 00   0f 08 00 00   31 08 00 00   33 08 00 00'

from 'tests/recs/de-13.34.aoe2record'
s?  p           x1        y1       x2       y2       0       (building_id)
01 01    00 2d 00 28 00 32 00  1f      00            48 00          0000ffffffff00010000a5220000'

all coordinates are padded with an int8 0. I'm not sure what the purpose of this.

I assume it's for larger map size coordinates which can't fit into int8? (but the coordinates themselves are still int8 unless we assume that the 0 before building id is part of y2 and shift them all to right).

All the other action types (build, attack, move) use floats for coordinates so this seems to be specific to WALL.

Comment (qwyt) [reply mgz/fast/__init__.py:183]: I guess the best solution would be to  have two different format string based on data length?

File: mgz/fast/__init__.py [hunk 3]
```diff
@@ -178,8 +180,28 @@ def parse_action(action_type, data):
         object_id, = struct.unpack_from('<3xI', data)
         return dict(object_ids=[object_id])
     if action_type == Action.WALL:
-        player_id, x, y = struct.unpack_from('<x3b', data)
-        return dict(player_id=player_id, x=x, y=y)
+
```
Comment (qwyt) [inline mgz/fast/__init__.py]: @happyleavesaoc new changes

**Pull request merged**